lib/metrics/outbox.js

```diff
@@ -0,0 +1,189 @@
+    for (const tenant in statistics) {
+      for (const [serviceName, stats] of Object.entries(statistics[tenant])) {
+        const now = Date.now()
+        const observationAttributes = {
+          'sap.tenancy.tenant_id': tenant,
+          'outbox.name': serviceName
+        }
+
+        batchResult.observe(observables.coldEntries, stats.coldEntries, observationAttributes)
+
+        batchResult.observe(observables.remainingEntries, stats.remainingEntries, observationAttributes)
+
+        const minStorageTimeSeconds = stats.minTimestamp ? Math.floor((now - new Date(stats.minTimestamp)) / 1000) : 0
+        batchResult.observe(observables.minStorageTimeSeconds, minStorageTimeSeconds, observationAttributes)
+
+        const medStorageTimeSeconds = stats.medTimestamp ? Math.floor((now - new Date(stats.medTimestamp)) / 1000) : 0
+        batchResult.observe(observables.medStorageTimeSeconds, medStorageTimeSeconds, observationAttributes)
+
+        const maxStorageTimeSeconds = stats.maxTimestamp ? Math.floor((now - new Date(stats.maxTimestamp)) / 1000) : 0
+        batchResult.observe(observables.maxStorageTimeInSeconds, maxStorageTimeSeconds, observationAttributes)
+
+        batchResult.observe(observables.incomingMessages, stats.incomingMessages, observationAttributes)
+
+        batchResult.observe(observables.outgoingMessages, stats.outgoingMessages, observationAttributes)
+      }
+    }
+  }, Object.values(observables))
+}
+
+function initTenantOutboxStatistics(statistics, tenant, outboxedServiceName) {
+  if (!statistics[tenant]) statistics[tenant] = {}
+  if (!statistics[tenant][outboxedServiceName]) {
+    statistics[tenant][outboxedServiceName] = {
+      incomingMessages: 0,
+      outgoingMessages: 0,
+      coldEntries: 0,
+      remainingEntries: 0,
+      minTimestamp: null,
+      medTimestamp: null,
+      maxTimestamp: null
+    }
+  }
+}
+
+module.exports = () => {
+  // Skip setup if outbox telemetry is disabled
+  if (!cds.env.requires.telemetry.metrics?._outbox) return
+
+  /** @type {Record<[key: string], Record<[key: string], object>>} */
+  const statistics = {}
+  const registeredServics = new Set()
+
+  cds.on('listening', () => {
+    const queueEntity = cds.model.definitions[PERSISTENT_QUEUE_DB_NAME]
+
+    initOutboxObservation(statistics)
+
+    // Register service when it's first found to be the target of an outboxed message
+    cds.db.after(['CREATE'], queueEntity, async (_, req) => {
+      const outboxedServiceName = req.data.target
+
+      if (!statistics[req.tenant]?.[outboxedServiceName]) {
+        // Enable statistics collection per outbox per tenant
+        initTenantOutboxStatistics(statistics, req.tenant, outboxedServiceName)
+
+        const unboxedService = cds.services[outboxedServiceName]
+        const outboxedService = cds.outboxed(unboxedService)
+
+        // REVISIT: Defaulting is required for cds^8 where the service does not expose 'maxAttempts'
```

Comment:
`outboxed` is cds^9 lingo. cds^8 it is `outbox`. but i think we can support with cds^9 only to make the coding clearer... then, we should consistently use `queue(d)` instead of `outbox(ed)`